query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/QueryRequestToPinotSQLConverter.java

```diff
@@ -40,14 +42,15 @@
   private final ViewDefinition viewDefinition;
   private final PinotFunctionConverter functionConverter;
   private final Joiner joiner = Joiner.on(", ").skipNulls();
+  private Map<String,Integer> selectionMap = new HashMap<String,Integer>();
```

Comment:
this class is stateless, this looks like it's tied to a specific request.

Reply:
yup, no need to make it a class level var, if at all it is needed, it can be instantiated by the method.

Reply:
updated the code